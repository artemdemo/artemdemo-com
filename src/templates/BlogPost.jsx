import React from 'react';
import { graphql } from 'gatsby';
import _get from 'lodash/get';
import Layout from '../components/Layout/Layout';
import BinaryPagination from '../components/Pagination/BinaryPagination';
import PostDate from '../components/PostDate/PostDate';
import * as utils from '../services/utils';

import 'prismjs/themes/prism.css';

class BlogPost extends React.Component {
    renderPagination() {
        const { previous, next } = this.props.pageContext;

        const prevSlug = _get(previous, 'fields.slug', null);
        const nextSlug = _get(next, 'fields.slug', null);

        return (
            <BinaryPagination
                previous={{
                    slug: prevSlug ? utils.prefixBlog(prevSlug) : null,
                    title: _get(previous, 'frontmatter.title', null),
                }}
                next={{
                    slug: nextSlug ? utils.prefixBlog(nextSlug) : null,
                    title: _get(next, 'frontmatter.title', null),
                }}
            />
        );
    }

    render() {
        const post = this.props.data.markdownRemark;
        const siteTitle = _get(this.props, 'data.site.siteMetadata.title');
        const siteDescription = post.excerpt;

        return (
            <Layout
                title={`${post.frontmatter.title} | ${siteTitle}`}
                description={siteDescription}
                metaTitle={siteTitle}
                location={this.props.location}
            >
                <h1>{post.frontmatter.title}</h1>
                <PostDate>
                    {post.frontmatter.date}
                </PostDate>

                <div
                    dangerouslySetInnerHTML={{__html: post.html}}
                />

                <hr />

                {this.renderPagination()}
            </Layout>
        );
    }
}

export default BlogPost;

export const pageQuery = graphql`
  query BlogPostBySlug($slug: String!) {
    site {
      siteMetadata {
        title
        author
      }
    }
    markdownRemark(fields: { slug: { eq: $slug } }) {
      id
      excerpt
      html
      frontmatter {
        title
        date(formatString: "MMMM DD, YYYY")
      }
    }
  }
`;
