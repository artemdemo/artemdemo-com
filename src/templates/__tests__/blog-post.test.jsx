import React from 'react';
import renderer from 'react-test-renderer';
import BlogPost from '../BlogPost';

jest.mock('gatsby');
jest.mock('../../components/Container/Container');
jest.mock('../../components/Pagination/Pagination');

describe('BlogPost', () => {
    it('should render', () => {
        const tree = renderer.create(
            <BlogPost
                data={{
                    markdownRemark: {
                        frontmatter: {
                            title: 'FrontMatter title',
                            date: '23 May, 2018'
                        },
                        excerpt: 'Post excerpt',
                        html: 'HTML content',
                    },
                    site: {
                        siteMetadata: { title: 'Site title' },
                    },
                }}
                location={{}}
                pageContext={{
                    previous: {
                        fields: { slug: '/prev' },
                        frontmatter: { title: 'Prev title' },
                    },
                    next: {
                        fields: { slug: '/next' },
                        frontmatter: { title: 'Next title' },
                    },
                }}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});