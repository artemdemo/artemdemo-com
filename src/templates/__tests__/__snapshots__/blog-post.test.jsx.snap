// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost should render 1`] = `
<div
  data-mock="Container"
>
  <div
    data-mock="Helmet"
  >
    {
  "htmlAttributes": {
    "lang": "en"
  },
  "meta": [
    {
      "name": "description",
      "content": "Post excerpt"
    }
  ],
  "title": "FrontMatter title | Site title"
}
  </div>
  <h3>
    <div
      data-mock="Link"
    >
      Site title
    </div>
  </h3>
  <h1>
    FrontMatter title
  </h1>
  <p
    style={
      Object {
        "display": "block",
        "marginBottom": "10px",
      }
    }
  >
    23 May, 2018
  </p>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "HTML content",
      }
    }
  />
  <hr />
  <p>
    Written by 
    <strong>
      Artem Demo
    </strong>
     who lives and works in Tel Aviv building useful things.
     
    <a
      href="https://github.com/artemdemo"
    >
      You should follow him on Github
    </a>
  </p>
  <div
    data-mock="BinaryPagination"
  >
    {
  "previous": {
    "slug": "/blog/prev",
    "title": "Prev title"
  },
  "next": {
    "slug": "/blog/next",
    "title": "Next title"
  }
}
  </div>
</div>
`;
